testing/metaopt/test_automl_sequential_wrapper.py

```diff
@@ -40,3 +37,43 @@ def test_automl_sequential_wrapper(tmpdir):
     # Then
     actual_mse = ((predicted_outputs - expected_outputs) ** 2).mean()
     assert actual_mse < mse_before
+
+
+def test_automl_sequential_wrapper_with_validation_split_wrapper(tmpdir):
+    # Given
+    data_inputs = np.array(range(100))
+    expected_outputs = np.array(range(100, 200))
+
+    hyperparameter_space = HyperparameterSpace({
+        'multiplication_1__multiply_by': RandInt(1, 3),
+        'multiplication_2__multiply_by': RandInt(1, 3),
+        'multiplication_3__multiply_by': RandInt(1, 3),
+    })
+
+    pipeline = Pipeline([
+        ('multiplication_1', MultiplyByN()),
+        ('multiplication_2', MultiplyByN()),
+        ('multiplication_3', MultiplyByN())
+    ], cache_folder=tmpdir).set_hyperparams_space(hyperparameter_space)
+
+    random_search = RandomSearch(
+        pipeline,
+        hyperparams_repository=HyperparamsJSONRepository(tmpdir),
+        validation_technique=ValidationSplitWrapper(
+            test_size=0.2,
+            scoring_function=mean_squared_error,
+            run_validation_split_in_test_mode=False
+        ),
+        n_iters=100,
+        higher_score_is_better=False
+    )
```

Comment:
Is it normal that I'm surprised here? I had this in mind to be able to pass any AutoML algorithm to a single wrapper rather than call the algorithm: 

```
    random_search = AutoMLSequentialWrapper(
        pipeline,
        RandomSearch(
            n_iters=100,
            higher_score_is_better=False
        ), 
        hyperparams_repository=HyperparamsJSONRepository(tmpdir),
        validation_technique=ValidationSplitWrapper(
            test_size=0.2,
            scoring_function=mean_squared_error,
            run_validation_split_in_test_mode=False
        ),
    )
```

Note that here I may have nested objects in a weird way that doesn't work, but the important thing to notice is that the `RandomSearch` is an argument of the constructor of `AutoMLSequentialWrapper`. That's it. 

This allows for changing the wrapper in the future independently from the algo. E.g.: 
1. Someone may change the wrapper, such as `AutoMLSequentialWrapper` by a `ClusterComputerParallelWrapper`. 
2. Someone may otherwise change the algo, such as keeping `AutoMLSequentialWrapper` but just changing the argument of the AutoML algo to `HyperoptTPE` for instance.

Reply:
EDIT: Forget what I just said. I checked again, and I saw this in the constructor of the `RandomSearch`: 

```
        AutoMLSequentialWrapper.__init__(
            self,
            step=step,
            auto_ml_algorithm=AutoMLAlgorithm(
                hyperparameter_optimizer=RandomSearchHyperparameterOptimizer(),
                validation_technique=validation_technique,
                higher_score_is_better=higher_score_is_better
            ),
            hyperparams_repository=hyperparams_repository,
            n_iters=n_iters
        )
```

So it is indeed possible to do as I thought. Phew. 

What I'm tempted to do now: 
1. Delete RandomSearch
2. Rename RandomSearchHyperparameterOptimizer to RandomSearch

This way, we won't have to create the wrapper again for each algorithm we create. 

I also suggest that we rename `AutoMLSequentialWrapper` to `SequentialAutoMLLoop` because from here every object will wrap another one anyway so using Wrapper is too generic and doesn't convey any helpful deep meaning. So we would later have `ClusteringParallelAutoMLLoop` and `SaverParallelAutoMLLoop`. 

Other comment: I myself don't like the word `Saver` in the name `SaverParallelAutoMLLoop` but that's a problem for later. I neither like the name `SaverParallelTransform`, but that's another issue.

Reply:
removed the deprecated random search from the random package. there is still one test left to fix before merging. removing the old random search broke one unit test.

Reply:
I skipped the test for merging this. To fix later, test marked as skipped.